src/components/Executions/Tables/WorkflowExecutionsTable.tsx

```diff
@@ -197,44 +260,21 @@ export const WorkflowExecutionsTable: React.FC<WorkflowExecutionsTableProps> = p
     // Custom renderer to allow us to append error content to executions which
     // are in a failed state
     const rowRenderer: ListRowRenderer = rowProps => {
-        const { index: rowIndex, style } = rowProps;
-        const execution = executions[rowIndex];
-        const { error } = execution.closure;
-
+        const execution = executions[rowProps.index];
+        const cacheKey = getCacheKey(execution.id);
+        const onExpandCollapseError = (expanded: boolean) => {
+            setExpandedErrors({ ...expandedErrors, [cacheKey]: expanded });
```

Comment:
We probably want to use the callback version of `setExpandedErrors` when merging objects to ensure we aren't utilizing stale state 

```suggestion
            setExpandedErrors(currentExpandedErrors => ({ ...currentExpandedErrors, [cacheKey]: expanded }));
```